### PR TITLE
add Workflow component

### PR DESCRIPTION
**Pull request by iamvigneshwars**
depends #158

### CODE REVIEW

File: frontend/workflows-lib/stories/WorkflowAccordion.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import type { Meta, StoryObj } from "@storybook/react";
+import WorkflowAccordion from "../lib/components/workflow/WorkflowAccordian";
+import { WorkflowStatus } from "../lib/types";
+
+const fakeTasks = [
```
Comment (garryod) [inline frontend/workflows-lib/stories/WorkflowAccordion.stories.tsx:5]: Could we move these to a common TS / JSON file?

File: frontend/workflows-lib/lib/types.ts [hunk 2]
```diff
@@ -8,3 +8,17 @@ export interface Task {
...
   children?: TaskNode[];
 }
+
+export enum WorkflowStatus {
+  Pending = "Pending",
+  Running = "Running",
+  Succeeded = "Succeeded",
+  Failed = "Failed",
+  Errored = "Errored",
+}
```
Comment (garryod) [inline frontend/workflows-lib/lib/types.ts:18]: Would it be better to use a union type (e.g. `type WorkflowStatus = "Pending" | "Running"`) instead?

File: frontend/workflows-lib/stories/WorkflowAccordion.stories.tsx (hunk 1 above)
Comment (iamvigneshwars) [reply frontend/workflows-lib/stories/WorkflowAccordion.stories.tsx:5]: That could be a separate PR? we could move it for all stories?

**Review by garryod — APPROVED** [review]
LGTM

**Pull request merged**